contrib/pkg/sets/sets.go

```diff
@@ -34,18 +34,14 @@ func Key(id ezkube.ResourceId) string {
 	if id == nil {
 		return "<unknown>"
 	}
-	if clusterId, ok := id.(ezkube.ClusterResourceId); ok {
-		b.WriteString(clusterId.GetName())
-		b.WriteString(separator)
-		b.WriteString(clusterId.GetNamespace())
-		b.WriteString(separator)
-		b.WriteString(clusterId.GetClusterName())
-		return b.String()
-	}
 	b.WriteString(id.GetName())
 	b.WriteString(separator)
 	b.WriteString(id.GetNamespace())
 	b.WriteString(separator)
+	if clusterId, ok := id.(ezkube.ClusterResourceId); ok {
+		b.WriteString(separator)
+		b.WriteString(ezkube.GetClusterName(clusterId))
+	}
```

Comment:
What if something is passed in here that fulfills the old interface? e.g. `GetClusterName()`

Reply:
As it stands, only the clusterName annotation would be checked, which would result in the clusterName effectively not propagating. So I should probably add another helper function in ezkube package to handle that case too.